config/Dockerfiles/linchpin-tests.sh

```diff
@@ -1,30 +1,49 @@
 #!/bin/bash -x
 
-base_dir="$( pwd )"
+base_dir="${PWD}"
 
 set -o pipefail
 
 mkdir -p ${base_dir}/${distro}_logs
 
+TEST_DIR="./config/Dockerfiles/tests.d"
+
+if [ -z "${PROVIDERS}" ]; then
+    PROVIDERS="dummy openstack"
+fi
+
+echo "Testing providers: ${PROVIDERS}"
+
 result=0
-for i in $TARGETS; do
-    testname="$distro/$i"
-    export CREDS_PATH="$base_dir/keys/$i"
+for provider in $PROVIDERS; do
+    export CREDS_PATH="$base_dir/keys/${provider}"
     # If CREDS_PATH provides a tarball extract it and
     # run it's install script
-    if [ -e "$CREDS_PATH/${i}.tgz" ]; then
+    if [ -e "$CREDS_PATH/${provider}.tgz" ]; then
         tmpdir=$(mktemp -d)
-        tar xvf $CREDS_PATH/${i}.tgz -C $tmpdir
+        tar xvf $CREDS_PATH/${provider}.tgz -C $tmpdir
         $tmpdir/install.sh
     fi
-    ./config/Dockerfiles/linchpin-test.sh $i ${distro} 2>&1 |tee ${distro}_logs/${i}.log
-    if [ $? -eq 0 ]; then
-        test_summary="$(tput setaf 2)SUCCESS$(tput sgr0)\t${testname}"
-    else
-        test_summary="$(tput setaf 1)FAILURE$(tput sgr0)\t${testname}\tlog: logs/${distro}_${i}.log"
-        result=1
-    fi
-    summary="${summary}\n${test_summary}"
+    > ${distro}_logs/${provider}.log
+    for testfile in ${TEST_DIR}/${provider}/*; do
```

Comment:
I was hoping we could have generic tests that would be run against every provider.  If the test didn't make sense for that provider you would use the ${distro}-${provider}.ignore file.